apiserver/README.md

```diff
@@ -1,81 +1,141 @@
+export PATH=$PATH:$GOPATH/bin
 ```
 
-### Configuration
+## Clone and build the API server
 
-The apiserver.conf is the primary configuration file. If running the apiserver outside of Docker, you will need to manually configure the following settings:
+Clone this repo:
+```
+mkdir -p $GOPATH/src/github.com/
+cd $GOPATH/src/github.com/
+git clone https://github.com/nds-org/ndslabs 
+```	
 
+Build local binaries:
+```
+./build.sh local
 ```
-[Server]
-Port=30001
-Origin=http://<CORS origin host>
-VolDir=/home/core/apiserver/volumes
-Host=<your host ip>
-VolumeSource=local
-SpecsDir=<optional path to local specs directory>
 
-[Etcd]
-Address=localhost:4001
+## Configure API server
+
+Clone the `ndslabs-specs` repo:
+```
+git clone https://github.com/nds-org/ndslabs-specs
+```
+
+Modify the `apiserver.json` to reflect your local configuration. At a minimum,
+set the `supportEmail` to your email address, `kubernetes.address` to 
+`https://minikube-ip:8443` and the `specs.path` to the location of the cloned repo above.
+
 
-[Kubernetes]
-Address=localhost:8080
+## Run the API server
 
+Assuming `minikube` and `etcd` are running, you can simply run the apiserver
+binary:
+
+```
+./build/bin/apiserver-darwin-amd64
 ```
 
-If VolumeSource is "local", a local directory is used for hostPath volumes in Kubernetes. 
+## Running integration tests
 
+Install `newman`:
+```
+npm install -g newman
+```
 
-### Running the server
+To configure the postman environment, edit `workbench.postman_environment.json`
+and set the `host` value to `localhost:30001` and `email` value to your email
+address.
 
-The server requires etcd and Kubernetes. You can either run a separate etcd or use the Kubernetes instance.
+Because the API server is running insecure, you'll need to edit `Workbench.postman_collection.json` 
+and replace all instances of `https` with `http`.
 
-To run
+Run the tests:
 ```
-./apiserver -v <log verbosity 1-4> --conf <path to apiserver.conf> --passwd <admin password>
+newman run --insecure  --environment=workbench.postman_environment.json --delay-request=1000 Workbench.postman_collection.json
+```
+
+## Running as an external service
+
+It is also possible to run the Workbench API server outside of the cluster. This has been tested with minikube via VirtualBox on MacOS.
+
+Create a file `external-apiserver.yaml` with the following. Note that 10.0.2.2 is the internal address of your minikube VM. You can confirm this via `minikube ssh` and `netstat -rn`:
 ```
+kind: "Service"
+apiVersion: "v1"
+metadata:
+  name: "ndslabs-apiserver"
+spec:
+  ports:
+    - protocol: "TCP"
+      port: 30001
+      targetPort: 30001
+---
+kind: "Endpoints"
+apiVersion: "v1"
+metadata:
+  name: "ndslabs-apiserver"
+subsets:
+  - addresses:
+    - ip: "10.0.2.2"
+    ports:
+    - port: 30001
+```
+
+Delete the in-cluster API server and create this external service/endpoint:
+```
+kubectl delete svc ndslabs-apiserver
+kubectl create -f external-apiserver.yaml
+```
```

Comment:
Has this configuration been tested with RBAC enabled? I'm wondering how the "external" might aspect affect our use of the Kubernetes API, but I may just be overthinking things...

Reply:
I haven't actually tested this -- it was from the original documentation.